var-test-support/src/main/java/io/varhttp/test/VarClient.java

```diff
@@ -11,4 +15,34 @@ public interface VarClient {
 	VarClientRequest get(String path);
 	VarClientRequest delete(String path);
 	VarClientRequest head(String path);
+
+	static HttpResponse getHttpResponseWithContent(VarClientRequest.ContentFormat contentFormat, HttpURLConnection conn) throws IOException {
+		HttpResponse httpResponse = new HttpResponse();
+
+		switch (contentFormat) {
+			case STREAM_CONTENT:
+				httpResponse.setInputStream(HttpClient.readDownloadableContent(conn));
```

Comment:
maybe this HttpClient can receive the response and populate it with this logic, and then we can just make this two HttpClient.readDownloadableContent(conn) and HttpClient.readContent(conn) private, so you always have to send what type of Content you want the client to resolve for you